resources/lib/modules/bluetooth.py

```diff
@@ -677,18 +678,18 @@ def stopped(self, value):
     @log.log_function()
     def stop(self):
         self.stopped = True
-        oe.dictModules['bluetooth'].stop_discovery()
+        self.parent.stop_discovery()
 
     @log.log_function()
     def run(self):
         self._stop_event.clear()
         while not self.stopped and not oe.xbmcm.abortRequested():
             current_time = time.time()
-            if current_time > self.last_run + 5:
-                if self.main_menu.getSelectedItem().getProperty('modul') != 'bluetooth' or not hasattr(oe.dictModules['bluetooth'], 'discovery_thread'):
-                    oe.dictModules['bluetooth'].menu_connections()
+            if self.main_menu.getSelectedItem().getProperty('modul') == 'bluetooth':
+                if current_time > self.last_run + 5:
+                    self.parent.discover_devices()
                 self.last_run = current_time
```

Comment:
Is this correct? IMO self.parent.discover_devices() will only be executed once at start.
```suggestion
                    self.last_run = current_time
```

Reply:
You are correct. I have changed the condition.